usaspending_api/common/helpers.py

```diff
@@ -94,32 +95,62 @@ def generate_last_completed_fiscal_quarter(fiscal_year):
 def get_pagination(results, limit, page, benchmarks=False):
     if benchmarks:
         start_pagination = time.time()
-    page_metadata = {"page": page, "count": len(results), "next": None, "previous": None, "hasNext": False,
-                     "hasPrevious": False}
+    page_metadata = {
+        "page": page,
+        "count": len(results),
+        "next": None,
+        "previous": None,
+        "hasNext": False,
+        "hasPrevious": False
+    }
     if limit < 1 or page < 1:
         return [], page_metadata
-    page_metadata["hasNext"] = (limit*page < len(results))
-    page_metadata["hasPrevious"] = (page > 1 and limit*(page-2) < len(results))
+
+    page_metadata["hasNext"] = (limit * page < len(results))
+    page_metadata["hasPrevious"] = (page > 1 and limit * (page - 2) < len(results))
+
     if not page_metadata["hasNext"]:
-        paginated_results = results[limit*(page-1):]
+        paginated_results = results[limit * (page - 1):]
     else:
-        paginated_results = results[limit*(page-1):limit*page]
-    page_metadata["next"] = page+1 if page_metadata["hasNext"] else None
-    page_metadata["previous"] = page-1 if page_metadata["hasPrevious"] else None
+        paginated_results = results[limit * (page - 1): limit * page]
+
+    page_metadata["next"] = page + 1 if page_metadata["hasNext"] else None
+    page_metadata["previous"] = page - 1 if page_metadata["hasPrevious"] else None
     if benchmarks:
         logger.info("get_pagination took {} seconds".format(time.time() - start_pagination))
     return paginated_results, page_metadata
 
 
 def get_pagination_metadata(total_return_count, limit, page):
-    page_metadata = {"page": page, "count": total_return_count, "next": None, "previous": None, "hasNext": False,
-                     "hasPrevious": False}
+    page_metadata = {
+        "page": page,
+        "count": total_return_count,
+        "next": None,
+        "previous": None,
+        "hasNext": False,
+        "hasPrevious": False
+    }
     if limit < 1 or page < 1:
         return page_metadata
+
     page_metadata["hasNext"] = (limit * page < total_return_count)
-    page_metadata["hasPrevious"] = (page > 1 and limit*(page-2) < total_return_count)
-    page_metadata["next"] = page+1 if page_metadata["hasNext"] else None
-    page_metadata["previous"] = page-1 if page_metadata["hasPrevious"] else None
+    page_metadata["hasPrevious"] = (page > 1 and limit * (page - 2) < total_return_count)
+    page_metadata["next"] = page + 1 if page_metadata["hasNext"] else None
+    page_metadata["previous"] = page - 1 if page_metadata["hasPrevious"] else None
+    return page_metadata
+
+
+def get_simple_pagination_metadata(results_plus_one, limit, page):
```

Comment:
I remember there were instances where this function could be useful. Is there any reason we haven't used this instead?

Reply:
I just made this function. It is useful if the count query is too expensive. Since I kept count in spending_by_category I didn't utilize this function.

Reply:
So for some of the endpoints currently we aren't returning `count` contrary to the gist. This was removed as the front end doesn't use count for this endpoint. I'd suggest double checking with the frontend team to ensure that is the case, update the gist accordingly, and use the simple_page_metadata solution. If the `count`s aren't as taxing as we thought they were, we can readd them back in but we made sure to remove them to save on performance and the frontend wasn't using them.

Reply:
Verified front end isn't relying on them. Updated gist [here](https://hackmd.io/s/SyIK1HcRb)